### ISSUE TITLE
🚀 Feature: automate upgrade process

### ISSUE THREAD

**Issue by Matanga1-2** [galezra/lusha-register-extension#27]
# 🔖 Feature Description
Given a selected plan/price/version, automatically purchase the plan and complete the process.

## 📝 Description
Today, when we want to upgrade an account, we need to -
* Go to the pricing page
* Select the plan
* Fill in all the details on the checkout page
* Complete the purchase
* See the receipt and proceed back to the dashboard

## ✌️ Implementation
There are a couple of challenges here, but the main lines should be -
* Prompt to get the desired plan/version (e.g., "4.2, professional, monthly, two users"). 
It could be a new tab on the configuration form.
* Show an "Upgrade" button when on the home screen. When you click, you can perform the upgrade.
* Fetch the wanted plan from the catalog and build the payment link (challenge - how to get the links)
* Go to checkout and fill in the details (challenge - Stripe card number is an iFrame, can't directly fill it)
* Click "Confirm," and once again, "Go to Dashboard" on the receipt page 

## 🔄️ Additional Info
Performing this action manually can take around 2-3 minutes. Automating it can allow a more flexible and efficient experience.

**Comment by galezra** [comment]
Your idea to streamline the upgrade process could greatly enhance user experience, despite the discussed implementation challenges. We will consult with our lead developer, Evgeni, to explore feasibility and solutions.

Your input is highly valued, let's keep this conversation going. We'll update this thread as we make progress. Thank you!

**Comment by Ivgesha** [comment]
This idea presents a promising initiative, and we can initiate the process by utilizing [Mermaid - Sequence diagrams](https://mermaid.js.org/syntax/sequenceDiagram.html) to visually represent the workflow and comprehensively assess all potential use cases.

**Comment by Matanga1-2** [comment]
Let's start by checking if we can automate the process on the checkout page as a first phase.
@galezra I'll attempt to solve this.

**Comment by galezra** [comment]
@Ivgesha I agree that visualizing the concept through a diagram would be highly beneficial. It will help us better understand the workflow and the integration points needed for implementation. If you could initiate the diagram or share your thoughts on how you envision it, that would be great!

@Matanga1-2 Sounds great! I think for a start, let's develop a vanilla script similar to the segmentation quiz that will run automatically after clicking on the 'Lugister' button, similar to the signup page. However, maybe it would be good to add a billing tab for configuration on the check page. what do you think?

**Comment by galezra** [comment]
You rock, @Ivgesha!
After reviewing the previous discussions and your latest suggestion, I am excited about the potential of your idea. Let me outline my thoughts on how we can move forward with it:

Configuration Page Layout: We can design a layout that includes tabs, a form area, and 'Save' + 'Cancel' buttons. The form content will dynamically change based on the selected tab, providing a seamless and user-friendly experience.

Infrastructure for Tabs and Forms: To make this work, we'll need to develop a robust infrastructure. This includes creating the tabs functionality and a mechanism to inject different forms based on the selected tab. This approach will allow us to keep the configuration page clean and organized.

Refining the Checkout Page Form: The checkout form page needs to be integrated with the useForm function. Also, the 'Save' button should effectively save the changes made in the current form tab. Ensuring that these elements work harmoniously is crucial for a smooth user experience.

Vanilla Script for Form Injection: Finally, we'll create a vanilla script that can inject the configuration form into the checkout page. This script should also enable the Lugister button to fill out the form with a single click. Additionally, if the auto-save feature is enabled, the form should be filled out automatically.

This plan essentially breaks down into three parts: developing the infrastructure, refining the checkout page form, and creating the script for form injection.

Note: I can take the tabs part or does someone else want to go with it?

Your feedback is invaluable, and I'm eager to hear your thoughts on this plan. Let's also loop in @Matanga1-2 for their insights.

Looking forward to collaborating on this!

**Comment by Matanga1-2** [comment]
> @Matanga1-2 Sounds great! By the way, why do we need Stripe if we just have to fill out the credit card information?

The card number is not on our end, it's in Stripe iFrame. Need to see how we can identify this component and if we can fill it properly
<img width="1300" alt="image" src="https://github.com/galezra/lusha-register-extension/assets/24847398/98ecc726-de2c-42dd-9177-fde75b7bf0b6">

**Comment by Matanga1-2** [comment]
> @Matanga1-2 After conducting a preliminary analysis of the project, I have developed an initial checkout form as a starting point (refer to my pull request above). I trust that you can build upon this foundation and further enhance its functionality.

Thank you @Ivgesha I just opened a new draft PR after merging some of your changes into my branch. Much appreciated!

**Comment by Matanga1-2** [comment]
@galezra @Ivgesha taking @Ivgesha contribution so far, I created this draft PR - 
https://github.com/galezra/lusha-register-extension/pull/36

I'm a bit stuck with the changes required in the useForm() - I'm able to load the CheckoutPage successfully, but during runtime it changes to contain the SignupForm fields.
I'm sure you can point me in the right direction to solving this.

For now my approach is -
1. Show Lugister button on checkout - DONE
2. Fill form upon click using default values - 80% done, need to solve the useForm() issue
3. If set as auto-signup, submit form - DONE, need to test after solving the useForm() issue
4. Fill in card number - I leave this to the end
5. Support dynamic values from the extension form - I suggest doing it as a 2nd upgrade in a different PR

**Comment by galezra** [comment]
Thanks a lot, @Matanga1-2 , for your fantastic work! I'm currently engaged in some infrastructure work related to the form mechanism. As you correctly pointed out, our current setup only supports signup forms. I'm in the process of separating the form and the key-value object, which will be structured as form: {signup: {...}, checkout: {...}}. This approach aims to facilitate support for numerous configurable form pages, allowing for a more organized and clean object structure for each page.

Once I complete the infrastructure work (I anticipate having a PR ready in a few hours), you can merge it into your branch. This will significantly simplify extending support for future form pages. After merging, we can continue to collaborate and tackle any further challenges together. 

note: we can definitely support dynamic values in the 2nd PR
